test/v5/connector_test.go

```diff
@@ -0,0 +1,483 @@
+// Copyright © 2022 Meroxa, Inc. and Miquido
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v5
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"testing"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	esV5 "github.com/elastic/go-elasticsearch/v5"
+	"github.com/jaswdr/faker"
+	"github.com/miquido/conduit-connector-elasticsearch/destination"
+	"github.com/miquido/conduit-connector-elasticsearch/internal/elasticsearch"
+	v5 "github.com/miquido/conduit-connector-elasticsearch/internal/elasticsearch/v5"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOperationsWithSmallestBulkSize(t *testing.T) {
```

Comment:
You mean `test/v5/connector_test.go`, `test/v6/connector_test.go` etc.? They are the same because they are testing the same logic across different versions of ES. Thanks to that, I detected a case where ES v5 did not support bulk operation in the same manner as others. I was thinking about extracting these tests logic and reusing it, but I left it for later. I'm not sure if it won't get more messy that way tbh.

Reply:
I definitely agree with that we should test all the versions which we claim to support. What I was curious about is the test code duplication. If the production code is not duplicated, then I believe the test code should not be duplicated either.

In other words, what I think would be great is to have a single set of tests, and run them against the different versions of ES.

Reply:
Gotcha, I also don't like this code to be repeated so extensively. I'll work on simplifying it 🙂

Reply:
I think that this too can go in a separate issue.

Reply:
Quick check: do we have an issue for the above?

Reply:
No, sorry for that! I have just created it: https://github.com/miquido/conduit-connector-elasticsearch/issues/6